storage/src/main/java/org/openjdk/skara/storage/FileStorage.java

```diff
@@ -64,6 +64,10 @@ public void put(Collection<T> items) {
             return;
         }
         try {
+            Files.createDirectories(file.getParent());
+            if (!Files.exists(file)) {
```

Comment:
Doesn't hurt, but Files.writeString should create it if it doesn't exist as well I think.

Reply:
It didn't for nested paths, so the `Files.createDirectories` is definitely needed. The `if (!Files.exists)` could potentially be left out, but otoh it doesn't hurt to be explicit here.